stablehlo/reference/Ops.cpp

```diff
@@ -1475,26 +1451,31 @@ Tensor evalSelectOp(const Tensor &pred, const Tensor &onTrue,
   return result;
 }
 
-Tensor evalSelectAndScatter(const Tensor &operand, const Tensor &source,
-                            const Tensor &initValue,
-                            const Sizes &windowDimensions,
-                            const Sizes &windowStrides, const Sizes &paddingLow,
-                            const Sizes &paddingHigh, Region &select,
-                            Region &scatter, Scope &scope,
-                            ShapedType resultType) {
+Tensor evalSelectAndScatterOp(const Tensor &operand, const Tensor &source,
+                              const Tensor &initValue,
+                              const Sizes &windowDimensions,
+                              const Sizes &windowStrides,
+                              const Sizes &paddingLow, Region &select,
+                              Region &scatter, Scope &scope,
+                              ShapedType resultType) {
   Tensor result(resultType, initValue.get({}));
-  auto rank = operand.getRank();
 
-  for (auto sourceIt = IndexSpaceIterator(source.getShape(), Sizes(rank, 0));
-       sourceIt != IndexSpaceIterator(source.getShape(), std::nullopt);
+  for (auto sourceIt = source.index_begin(); sourceIt != source.index_end();
```

Comment:
Not sure why this wasn't `source.index_begin()` and `source.index_end()` in the first place.